code/modules/shuttle/super_cruise/shuttle_components/shuttle_docking.dm

```diff
@@ -83,13 +90,19 @@
 			for(var/turf/T in A)
 				if(T.get_virtual_z_level() != origin.get_virtual_z_level())
 					continue
+				// if it has doors or something passible, turn on a flag so it can return a better flag to show better result
+				var/passible = FALSE
+				for(var/each in passible_things)
+					if(locate(each) in T)
+						passible = TRUE
+						break
 				var/image/I = image('icons/effects/alphacolors.dmi', origin, "red")
 				var/x_off = T.x - origin.x
 				var/y_off = T.y - origin.y
 				I.loc = locate(origin.x + x_off, origin.y + y_off, origin.z) //we have to set this after creating the image because it might be null, and images created in nullspace are immutable.
 				I.plane = ABOVE_LIGHTING_PLANE
 				I.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
-				the_eye.placement_images[I] = list(x_off, y_off)
+				the_eye.placement_images[I] = list(x_off, y_off, "passible"=passible)
```

Comment:
This will turn it from an indexed list into an assoc list which could be funky since 2 elements have no value. Either access passable by index (better performance and fully backwards compatible so less potential bug opportunity) or make all the fields have keys